src/Interpreters/executeQuery.cpp

```diff
@@ -259,15 +259,20 @@ addStatusInfoToQueryLogElement(QueryLogElement & element, const QueryStatusInfo
 
     /// We need to refresh the access info since dependent views might have added extra information, either during
     /// creation of the view (PushingToViews chain) or while executing its internal SELECT
-    const auto & access_info = context_ptr->getQueryAccessInfo();
+
+    /// We copy QueryAccessInfo for thread-safety, because it is possible that query context can be modified by some processor even
+    /// after query is finished
+    ContextData::QueryAccessInfo access_info(context_ptr->getQueryAccessInfo());
```

Comment:
I'm having a look at whether it'd be better to mark the members as `TSA_GUARDED_BY(mutex)` and force acquiring the mutex here.